File: print-manager-dotnet-core-vb/ReadPrinterDetails.vb
Imports ceTe.DynamicPDF.Printing

Public Class ReadPrinterDetails
    Public Shared Sub Run()
        Dim MyPrintJob As PrintJob = New PrintJob(Printer.Default)

        Console.WriteLine("Options for: " + MyPrintJob.Printer.Name)
        Console.WriteLine("Collate " + MyPrintJob.Printer.Collate.ToString())
        Console.WriteLine("Duplex " + MyPrintJob.Printer.Duplex.ToString())
        Console.WriteLine("MaxCopies " + MyPrintJob.Printer.MaxCopies.ToString())
        Console.WriteLine("MaxPaperHeight " + MyPrintJob.Printer.MaxPaperHeight.ToString())
        Console.WriteLine("MaxPaperWidth " + MyPrintJob.Printer.MaxPaperWidth.ToString())
        Console.WriteLine("MinPaperHeight " + MyPrintJob.Printer.MinPaperHeight.ToString())
        Console.WriteLine("MinPaperWidth " + MyPrintJob.Printer.MinPaperWidth.ToString())
        ListPaperSizes(MyPrintJob.Printer.PaperSizes)
        ListPaperSources(MyPrintJob.Printer.PaperSources)
        ListResolutions(MyPrintJob.Printer.Resolutions)
    End Sub

    Shared Sub ListPaperSizes(paperSizes As PaperSizeList)
        Console.WriteLine("List of Paper Sizes: ")
        For i As Integer = 0 To paperSizes.Count - 1
            Console.WriteLine(" " & paperSizes(i).Name & " : Width= " & UnitConverter.PointsToInches(paperSizes(i).Width.ToString()) & " : Height = " & UnitConverter.PointsToInches(paperSizes(i).Height.ToString()))
        Next
    End Sub

    Shared Sub ListPaperSources(paperSources As PaperSourceList)
        Console.WriteLine("List of Paper Sources ")
        For i As Integer = 0 To paperSources.Count - 1
            Console.WriteLine(" " & paperSources(i).Name)
        Next
    End Sub

    Shared Sub ListResolutions(resolutions As ResolutionList)
        Console.WriteLine("List of Resolution ")
        For i As Integer = 0 To resolutions.Count - 1
            Console.WriteLine(" " & resolutions(i).HorizontalDpi.ToString() & " dpi " & resolutions(i).VerticalDpi.ToString() & " dpi")
        Next
    End Sub
End Class

File: print-manager-dotnet-core-cs/ReadPrinterDetails.cs
using System;
using ceTe.DynamicPDF.Printing;

namespace print_manager_dotnet_core_cs
{
    class ReadPrinterDetails
    {
        public static void Run()
        {
            PrintJob printJob = new PrintJob(Printer.Default);

            Console.WriteLine("Options for: " + printJob.Printer.Name);
            Console.WriteLine("Collate " + printJob.Printer.Collate);
            Console.WriteLine("Duplex " + printJob.Printer.Duplex);
            Console.WriteLine("MaxCopies " + printJob.Printer.MaxCopies);
            Console.WriteLine("MaxPaperHeight " + printJob.Printer.MaxPaperHeight);
            Console.WriteLine("MaxPaperWidth " + printJob.Printer.MaxPaperWidth);
            Console.WriteLine("MinPaperHeight " + printJob.Printer.MinPaperHeight);
            Console.WriteLine("MinPaperWidth " + printJob.Printer.MinPaperWidth);
            ListPaperSizes(printJob.Printer.PaperSizes);
            ListPaperSources(printJob.Printer.PaperSources);
            ListResolutions(printJob.Printer.Resolutions);
        }

        static void ListPaperSizes(PaperSizeList paperSizes)
        {
            Console.WriteLine("List of Paper Sizes: ");
            for (int i = 0; i < paperSizes.Count; i++)
            {
                Console.WriteLine(" " + paperSizes[i].Name + " : Width= " + UnitConverter.PointsToInches(paperSizes[i].Width) + " : Height = " + UnitConverter.PointsToInches(paperSizes[i].Height));
            }
        }

        static void ListPaperSources(PaperSourceList paperSources)
        {
            Console.WriteLine("List of Paper Sources ");
            for (int i = 0; i < paperSources.Count; i++)
            {
                Console.WriteLine(" " + paperSources[i].Name);
            }
        }

        static void ListResolutions(ResolutionList resolutions)
        {
            Console.WriteLine("List of Resolution ");
            for (int i = 0; i < resolutions.Count; i++)
            {
                Console.WriteLine(" " + resolutions[i].HorizontalDpi + " dpi " + resolutions[i].VerticalDpi + " dpi");
            }
        }
    }
}
